packages/@lwc/ssr-runtime/src/stubs.ts

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+// Stubs for all the un-implemented exports from @lwc/engine-server
+
+export function api(..._: unknown[]): never {
+    throw new Error('@api cannot be used in SSR context.');
+}
+export function createContextProvider(..._: unknown[]): never {
+    throw new Error('createContextProvider cannot be used in SSR context.');
+}
+export function createElement(..._: unknown[]): never {
+    throw new Error('createElement cannot be used in SSR context.');
+}
+export function freezeTemplate(..._: unknown[]): never {
+    throw new Error('freezeTemplate cannot be used in SSR context.');
+}
+export function getComponentDef(..._: unknown[]): never {
+    throw new Error('getComponentDef cannot be used in SSR context.');
+}
+export function isComponentConstructor(..._: unknown[]): never {
+    throw new Error('isComponentConstructor cannot be used in SSR context.');
+}
+export function parseFragment(..._: unknown[]): never {
+    throw new Error('parseFragment cannot be used in SSR context.');
+}
+export function parseSVGFragment(..._: unknown[]): never {
+    throw new Error('parseSVGFragment cannot be used in SSR context.');
+}
+export function readonly(..._: unknown[]): never {
+    throw new Error('readonly cannot be used in SSR context.');
+}
+export function registerComponent(..._: unknown[]): never {
+    throw new Error('registerComponent cannot be used in SSR context.');
+}
+export function registerDecorators(..._: unknown[]): never {
+    throw new Error('registerDecorators cannot be used in SSR context.');
+}
+export function registerTemplate(..._: unknown[]): never {
+    throw new Error('registerTemplate cannot be used in SSR context.');
+}
+export function renderComponent(..._: unknown[]): never {
+    throw new Error('renderComponent cannot be used in SSR context.');
+}
```

Comment:
```suggestion
export function renderComponent(..._: unknown[]): never {
    throw new Error('renderComponent cannot be used in SSR context.');
}
```

This needs to be exported as an alias of `serverSideRenderComponent`. A test is failing without this.